openedx/core/djangoapps/content_libraries/rest_api/serializers.py

```diff
@@ -230,6 +231,45 @@ class LibraryXBlockStaticFilesSerializer(serializers.Serializer):
     files = LibraryXBlockStaticFileSerializer(many=True)
 
 
+class LibraryContainerMetadataSerializer(serializers.Serializer):
+    """
+    Serializer for Containers like Sections, Subsections, Units
+
+    Converts from ContainerMetadata to JSON-compatible data
+    """
+    container_key = serializers.CharField(read_only=True)
+    container_type = serializers.CharField()
+    display_name = serializers.CharField()
+    last_published = serializers.DateTimeField(format=DATETIME_FORMAT, read_only=True)
+    published_by = serializers.CharField(read_only=True)
+    last_draft_created = serializers.DateTimeField(format=DATETIME_FORMAT, read_only=True)
+    last_draft_created_by = serializers.CharField(read_only=True)
+    has_unpublished_changes = serializers.BooleanField(read_only=True)
+    created = serializers.DateTimeField(format=DATETIME_FORMAT, read_only=True)
+    modified = serializers.DateTimeField(format=DATETIME_FORMAT, read_only=True)
+    tags_count = serializers.IntegerField(read_only=True)
+    collections = CollectionMetadataSerializer(many=True, required=False, read_only=True)
+
+    # When creating a new container in a library, the slug becomes the ID part of
+    # the definition key and usage key:
+    slug = serializers.CharField(write_only=True)
```

Comment:
Can we leave the `slug` optional so the frontend can only send the `display_name` when creating?
```suggestion
    slug = serializers.CharField(write_only=True, required=False)
```

Reply:
Oops.. Already fixed! :smile:

Reply:
😄